sympy/solvers/solvers.py

```diff
@@ -442,7 +446,7 @@ def check_assumptions(expr, against=None, **assumptions):
     for key in a:
```

Comment:
Rather than compute them all first, pass them to `fuzzy_and` thru an iterator:
```
def _test(key):
    v = getattr(expr, 'is_' + key, None)
    if v is not None:
        return a[key] is v
return fuzzy_and(_test(k) for k in a)
```